DUI3-DX/Converters/Revit/Speckle.Converters.RevitShared/ToSpeckle/RoofBaseConversionToSpeckle.cs

```diff
@@ -0,0 +1,39 @@
+using Autodesk.Revit.DB;
+using Objects.BuiltElements.Revit.RevitRoof;
+using Speckle.Converters.Common;
+using Speckle.Converters.RevitShared.Helpers;
+
+namespace Speckle.Converters.RevitShared.ToSpeckle;
+
+[NameAndRankValue(nameof(DB.RoofBase), 0)]
+internal sealed class RoofBaseConversionToSpeckle : BaseConversionToSpeckle<DB.RoofBase, RevitRoof>
+{
+  private readonly DisplayValueExtractor _displayValueExtractor;
+  private readonly HostedElementConversionToSpeckle _hostedElementConverter;
+  private readonly ParameterObjectAssigner _parameterObjectAssigner;
+
+  public RoofBaseConversionToSpeckle(
+    DisplayValueExtractor displayValueExtractor,
+    HostedElementConversionToSpeckle hostedElementConverter,
+    ParameterObjectAssigner parameterObjectAssigner
+  )
+  {
+    _displayValueExtractor = displayValueExtractor;
+    _hostedElementConverter = hostedElementConverter;
+    _parameterObjectAssigner = parameterObjectAssigner;
+  }
+
+  public override RevitRoof RawConvert(RoofBase target)
+  {
+    RevitRoof revitRoof = new();
+    var elementType = (ElementType)target.Document.GetElement(target.GetTypeId());
```

Comment:
I've just made this change in the Floor PR too.

My understanding is that requesting an element in the document using it's `TypeId` will always return a `DB.ElementType` so this cast is safe. If it explodes by any means it would be an unexpected result.